src/TLuaInterpreter.cpp

```diff
@@ -12015,6 +12015,74 @@ int TLuaInterpreter::sendIrc( lua_State * L )
     return 0;
 }
 
+int TLuaInterpreter::setServerEncoding(lua_State * L)
+{
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if (! pHost) {
+        lua_pushstring(L, "setServerEncoding: NULL Host pointer - something is wrong!");
+        return lua_error(L);
+    }
+
+    QString newEncoding;
+    if (! lua_isstring(L, 1)) {
+        lua_pushfstring(L, "setServerEncoding: bad argument #1 type (newEncoding as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error( L );
+    }
+    else {
+        newEncoding = QString::fromUtf8(lua_tostring(L,1));
+    }
+
+    QPair<bool, QString> results = pHost->mTelnet.setEncoding(newEncoding);
+
+    if(results.first) {
+        lua_pushboolean(L, true);
```

Comment:
I'm aware it only shows once it is set to something else. 

So the logic is that for the rare case that it is changed incorrectly, we always pollute the display with the message and don't let the API do it themselves if they'd like.

What could be so confusing about the encoding suddenly becoming corrupt in the middle of the stream that we can't tell it was an encoding change? There is nothing else that could produce such an effect and debugging it would be as simple as searching for the `setServerEncoding()` in scripts.